src/hooks/useGenericLocation.ts

```diff
@@ -0,0 +1,8 @@
+import { Location } from "react-router-dom";
+
+export function useGenericLocation<T>(useLocation: Location) {
+  const location = useLocation;
+  const state: T = location.state;
+
+  return state;
+}
```

Comment:
1. 단순히 any에 T를 붙여주는 것만으로는 충분하지 않다고 생각해요. 이런 부분에서는 runtime validation이 필요하다고 생각합니다.
2. location을 굳이 전달받을 필요가 없어요. 훅 안에서 호출하면 되니까요.
3. 이런 커스텀 훅은 기존의 useLocation 훅을 대체하는 매우 low-level 한 훅이 됩니다. Type-safe 한 useLocation 이라는걸 이름에 드러냈으면 좋겠어요!


```suggestion
import { useLocation } from "react-router-dom";

export function useTypedLocation<T>(
  validate: (state: unknown) => state is T
) {
  const { state } = useLocation();

  if (!validate(state)) {
    throw new Error("Invalid state");
  }

  return state;
}
```